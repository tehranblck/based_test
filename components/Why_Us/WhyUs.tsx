import { Shield, Tag, Headphones, Clock } from "lucide-react";
import FeatureBox from "./FeatureBox";

export default function WhyUs() {
    const features = [
        {
            icon: <Shield className="w-6 h-6 text-gray-800" />,
            title: "Rəsmi Hesablar",
            description: "Sizə yalnız lisenziyalı, rəsmi və təhlükəsiz premium hesablar təqdim edirik. Saxta və qeyri-qanuni profillərlə riskə getməyin - Based.Az etibarinizi və hesab təhlükəsizliyinizi qoruyur."
        },
        {
            icon: <Tag className="w-6 h-6 text-gray-800" />,
            title: "Münasib Qiymətlər",
            description: "Ən populyar platformlara sərfəli qiymətlərlə çıxış imkanı veririk. Premium keyfiyyəti artıq baha qiyməte almaqa ehtiyac yoxdur - Based.Az ilə cibiniz də rahat nəfəs alacaq."
        },
        {
            icon: <Headphones className="w-6 h-6 text-gray-800" />,
            title: "24/7 Dəstək Komandası",
            description: "Sual və çətinlik yaranırsa, gecə və ya gündüz fərq etməz - Based.Az dəstək komandası daim yanınızdadır. Bizə yazın, tez və dəqiq kömək alacaqsınız."
        },
        {
            icon: <Clock className="w-6 h-6 text-gray-800" />,
            title: "Sürətli Təhvil",
            description: "Sifarişi etdiyiniz xidmətlər dərhal aktivləşdirilir. Uzun gözləmə yoxdur - Based.Az ilə sürətli və problemsiz keçid sizin olur."
        }
    ];

    return (
        <section className="py-16 px-4 bg-background">
            <div className=" mx-auto">
                <div className="text-center mb-12">
                    <h2 className="text-3xl md:text-4xl font-bold text-foreground mb-4">
                        Niyə Biz?
                    </h2>
                    <p className="text-lg text-muted-foreground max-w-2xl mx-auto">
                        "Rahatlıq, təhlükəsizlik və sərfəli qiymətlər bir arada."
                    </p>
                </div>

                <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-6">
                    {features.map((feature, index) => (
                        <FeatureBox
                            key={index}
                            icon={feature.icon}
                            title={feature.title}
                            description={feature.description}
                        />
                    ))}
                </div>
            </div>
        </section>
    );
}
