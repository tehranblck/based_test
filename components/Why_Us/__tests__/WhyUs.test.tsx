import { render, screen } from "@testing-library/react";
import WhyUs from "../WhyUs";

describe("WhyUs", () => {
    it("renders the section title and subtitle", () => {
        render(<WhyUs />);

        expect(screen.getByText("Niyə Biz?")).toBeInTheDocument();
        expect(screen.getByText('"Rahatlıq, təhlükəsizlik və sərfəli qiymətlər bir arada."')).toBeInTheDocument();
    });

    it("renders all four feature boxes", () => {
        render(<WhyUs />);

        expect(screen.getByText("Rəsmi Hesablar")).toBeInTheDocument();
        expect(screen.getByText("Münasib Qiymətlər")).toBeInTheDocument();
        expect(screen.getByText("24/7 Dəstək Komandası")).toBeInTheDocument();
        expect(screen.getByText("Sürətli Təhvil")).toBeInTheDocument();
    });

    it("renders feature descriptions", () => {
        render(<WhyUs />);

        expect(screen.getByText(/Sizə yalnız lisenziyalı, rəsmi və təhlükəsiz premium hesablar/)).toBeInTheDocument();
        expect(screen.getByText(/Ən populyar platformlara sərfəli qiymətlərlə çıxış imkanı/)).toBeInTheDocument();
        expect(screen.getByText(/Sual və çətinlik yaranırsa, gecə və ya gündüz fərq etməz/)).toBeInTheDocument();
        expect(screen.getByText(/Sifarişi etdiyiniz xidmətlər dərhal aktivləşdirilir/)).toBeInTheDocument();
    });

    it("applies correct section styling", () => {
        const { container } = render(<WhyUs />);
        const section = container.querySelector("section");

        expect(section).toHaveClass("py-16", "px-4", "bg-background");
    });

    it("has responsive grid layout", () => {
        render(<WhyUs />);
        const gridContainer = screen.getByText("Rəsmi Hesablar").closest(".grid");

        expect(gridContainer).toHaveClass("grid", "grid-cols-1", "md:grid-cols-2", "lg:grid-cols-4", "gap-6");
    });

    it("centers content with max width container", () => {
        const { container } = render(<WhyUs />);
        const mainContainer = container.querySelector(".max-w-7xl");

        expect(mainContainer).toBeInTheDocument();
        expect(mainContainer).toHaveClass("mx-auto");
    });

    it("renders title with correct typography", () => {
        render(<WhyUs />);
        const title = screen.getByText("Niyə Biz?");

        expect(title).toHaveClass("text-3xl", "md:text-4xl", "font-bold", "text-foreground", "mb-4");
    });

    it("renders subtitle with correct styling", () => {
        render(<WhyUs />);
        const subtitle = screen.getByText('"Rahatlıq, təhlükəsizlik və sərfəli qiymətlər bir arada."');

        expect(subtitle).toHaveClass("text-lg", "text-muted-foreground", "max-w-2xl", "mx-auto");
    });

    it("renders icons for all features", () => {
        const { container } = render(<WhyUs />);
        const icons = container.querySelectorAll("svg");

        expect(icons).toHaveLength(4);
    });

    it("has proper semantic HTML structure", () => {
        render(<WhyUs />);

        expect(screen.getByRole("heading", { level: 2 })).toHaveTextContent("Niyə Biz?");
    });

    it("renders exactly 4 feature boxes", () => {
        const { container } = render(<WhyUs />);
        const featureBoxes = container.querySelectorAll(".bg-card");

        expect(featureBoxes).toHaveLength(4);
    });

    it("has proper spacing for header section", () => {
        const { container } = render(<WhyUs />);
        const headerSection = container.querySelector(".text-center");

        expect(headerSection).toHaveClass("mb-12");
    });

    it("matches snapshot", () => {
        const { container } = render(<WhyUs />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
