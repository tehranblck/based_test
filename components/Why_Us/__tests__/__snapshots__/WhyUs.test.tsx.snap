// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`WhyUs matches snapshot 1`] = `
<section
  class="py-16 px-4 bg-background"
>
  <div
    class=" mx-auto"
  >
    <div
      class="text-center mb-12"
    >
      <h2
        class="text-3xl md:text-4xl font-bold text-foreground mb-4"
      >
        Niyə Biz?
      </h2>
      <p
        class="text-lg text-muted-foreground max-w-2xl mx-auto"
      >
        "Rahatlıq, təhlükəsizlik və sərfəli qiymətlər bir arada."
      </p>
    </div>
    <div
      class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-6"
    >
      <div
        class="group bg-card border border-border rounded-lg p-6 hover:shadow-xl hover:shadow-yellow-400/10 hover:-translate-y-1 hover:border-yellow-400/30 transition-all duration-300 ease-out cursor-pointer"
      >
        <div
          class="mb-4 inline-flex items-center justify-center w-12 h-12 bg-yellow-400 rounded-lg group-hover:scale-110 group-hover:rotate-3 group-hover:bg-yellow-300 transition-all duration-300 ease-out"
        >
          <div
            class="group-hover:scale-90 transition-transform duration-200 ease-out"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-shield w-6 h-6 text-gray-800"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 13c0 5-3.5 7.5-7.66 8.95a1 1 0 0 1-.67-.01C7.5 20.5 4 18 4 13V6a1 1 0 0 1 1-1c2 0 4.5-1.2 6.24-2.72a1.17 1.17 0 0 1 1.52 0C14.51 3.81 17 5 19 5a1 1 0 0 1 1 1z"
              />
            </svg>
          </div>
        </div>
        <h3
          class="text-lg font-semibold text-foreground mb-3 group-hover:text-yellow-600 dark:group-hover:text-yellow-400 transition-colors duration-200"
        >
          Rəsmi Hesablar
        </h3>
        <p
          class="text-sm text-muted-foreground leading-relaxed group-hover:text-foreground transition-colors duration-200"
        >
          Sizə yalnız lisenziyalı, rəsmi və təhlükəsiz premium hesablar təqdim edirik. Saxta və qeyri-qanuni profillərlə riskə getməyin - Based.Az etibarinizi və hesab təhlükəsizliyinizi qoruyur.
        </p>
      </div>
      <div
        class="group bg-card border border-border rounded-lg p-6 hover:shadow-xl hover:shadow-yellow-400/10 hover:-translate-y-1 hover:border-yellow-400/30 transition-all duration-300 ease-out cursor-pointer"
      >
        <div
          class="mb-4 inline-flex items-center justify-center w-12 h-12 bg-yellow-400 rounded-lg group-hover:scale-110 group-hover:rotate-3 group-hover:bg-yellow-300 transition-all duration-300 ease-out"
        >
          <div
            class="group-hover:scale-90 transition-transform duration-200 ease-out"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-tag w-6 h-6 text-gray-800"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12.586 2.586A2 2 0 0 0 11.172 2H4a2 2 0 0 0-2 2v7.172a2 2 0 0 0 .586 1.414l8.704 8.704a2.426 2.426 0 0 0 3.42 0l6.58-6.58a2.426 2.426 0 0 0 0-3.42z"
              />
              <circle
                cx="7.5"
                cy="7.5"
                fill="currentColor"
                r=".5"
              />
            </svg>
          </div>
        </div>
        <h3
          class="text-lg font-semibold text-foreground mb-3 group-hover:text-yellow-600 dark:group-hover:text-yellow-400 transition-colors duration-200"
        >
          Münasib Qiymətlər
        </h3>
        <p
          class="text-sm text-muted-foreground leading-relaxed group-hover:text-foreground transition-colors duration-200"
        >
          Ən populyar platformlara sərfəli qiymətlərlə çıxış imkanı veririk. Premium keyfiyyəti artıq baha qiyməte almaqa ehtiyac yoxdur - Based.Az ilə cibiniz də rahat nəfəs alacaq.
        </p>
      </div>
      <div
        class="group bg-card border border-border rounded-lg p-6 hover:shadow-xl hover:shadow-yellow-400/10 hover:-translate-y-1 hover:border-yellow-400/30 transition-all duration-300 ease-out cursor-pointer"
      >
        <div
          class="mb-4 inline-flex items-center justify-center w-12 h-12 bg-yellow-400 rounded-lg group-hover:scale-110 group-hover:rotate-3 group-hover:bg-yellow-300 transition-all duration-300 ease-out"
        >
          <div
            class="group-hover:scale-90 transition-transform duration-200 ease-out"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-headphones w-6 h-6 text-gray-800"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 14h3a2 2 0 0 1 2 2v3a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-7a9 9 0 0 1 18 0v7a2 2 0 0 1-2 2h-1a2 2 0 0 1-2-2v-3a2 2 0 0 1 2-2h3"
              />
            </svg>
          </div>
        </div>
        <h3
          class="text-lg font-semibold text-foreground mb-3 group-hover:text-yellow-600 dark:group-hover:text-yellow-400 transition-colors duration-200"
        >
          24/7 Dəstək Komandası
        </h3>
        <p
          class="text-sm text-muted-foreground leading-relaxed group-hover:text-foreground transition-colors duration-200"
        >
          Sual və çətinlik yaranırsa, gecə və ya gündüz fərq etməz - Based.Az dəstək komandası daim yanınızdadır. Bizə yazın, tez və dəqiq kömək alacaqsınız.
        </p>
      </div>
      <div
        class="group bg-card border border-border rounded-lg p-6 hover:shadow-xl hover:shadow-yellow-400/10 hover:-translate-y-1 hover:border-yellow-400/30 transition-all duration-300 ease-out cursor-pointer"
      >
        <div
          class="mb-4 inline-flex items-center justify-center w-12 h-12 bg-yellow-400 rounded-lg group-hover:scale-110 group-hover:rotate-3 group-hover:bg-yellow-300 transition-all duration-300 ease-out"
        >
          <div
            class="group-hover:scale-90 transition-transform duration-200 ease-out"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-clock w-6 h-6 text-gray-800"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 6v6l4 2"
              />
              <circle
                cx="12"
                cy="12"
                r="10"
              />
            </svg>
          </div>
        </div>
        <h3
          class="text-lg font-semibold text-foreground mb-3 group-hover:text-yellow-600 dark:group-hover:text-yellow-400 transition-colors duration-200"
        >
          Sürətli Təhvil
        </h3>
        <p
          class="text-sm text-muted-foreground leading-relaxed group-hover:text-foreground transition-colors duration-200"
        >
          Sifarişi etdiyiniz xidmətlər dərhal aktivləşdirilir. Uzun gözləmə yoxdur - Based.Az ilə sürətli və problemsiz keçid sizin olur.
        </p>
      </div>
    </div>
  </div>
</section>
`;
